web3.js/src/epoch-schedule.ts

```diff
@@ -0,0 +1,65 @@
+const MINIMUM_SLOT_PER_EPOCH = 32;
+
+// Returns the number of trailing zeros in the binary representation of self.
+function trailingZeros(n: number) {
+  let trailingZeros = 0;
+  while ((n & 1) == 0) {
+    n /= 2;
+    trailingZeros++;
+  }
+  return trailingZeros;
+}
+
+/**
+ * Epoch schedule
+ * (see https://docs.solana.com/terminology#epoch)
+ */
+export class EpochSchedule {
```

Comment:
It would be awesome to have `getEpoch(slot: number)` and `getEpochAndSlotIndex(slot: number)` too :)